### DISCUSSION TITLE
Is egg compatible with Python and machine learning?

### DISCUSSION THREAD

**Discussion by brando90** [egraphs-good/egg#157] [Q&A] [answered]
I wanted to build a project using E-graphs in Python with machine learning. Most tools in machine learning are in python so I am wondering how egg would be using here (or any e-graph really). Is egg compatible with python?

----

related: 
- https://github.com/ellisk42/ec/issues/89

**Comment by mwillsey** [comment, marked as answer]
No, you’d have to write Python <-> Rust bindings. Luckily, there are already some pretty flexible Python bindings! Check out https://github.com/egraphs-good/snake-egg
